# Exam 1 - Gema Castillo García

## Problem 2

Create a directory listing command that shows:

* the ownership of the file
* its file-size, in Megabytes (i.e. human readable)

then:

* in words (in a Markdown box), describe the _permissions_ on that file (read/write/execute) for users, groups, and "anyone"

In [2]:
ls -lh --author

total 1.5M
-rw-rw-r-- 1 osboxes osboxes osboxes 260K Sep  6 14:01 Exam_1_Answers.ipynb
-rw-rw-r-- 1 osboxes osboxes osboxes 1.1K Sep  6 13:12 LICENSE
-rw-rw-r-- 1 osboxes osboxes osboxes 1.2M Sep  3 15:51 Locus_Germplasm_Phenotype_20130122.txt
-rw-rw-r-- 1 osboxes osboxes osboxes   10 Sep  6 13:12 README.md


The first column shows the permissions on the files (I will explain it later) and the second one shows the number of hard links to these files (e.g., 1). Then, there are three columns that show the owner, group and author of the files, respectively. Finallly, there are the size and last modified date and time.

--author shows the column of the authors and -h shows file-size in Megabytes (i.e. human readable). Therefore, the ownership of the Locus_Germplasm_Phenotype_20130122.txt file is osboxes and its file-size is 1.2 Mb.


The column of permissions is structured like this: -/---/---/--- (file type/owner/groups/any user or program). Therefore, the permissions on the Locus_Germplasm_Phenotype_20130122.txt file are read (r) and write (w) for the owner and groups, and read (r) for anyone.






## Problem 3

* Create a command that outputs only the "header" line of Locus_Germplasm_Phenotype_20130122.txt

In [2]:
head -1 Locus_Germplasm_Phenotype_20130122.txt

Locus_name)	Germplasm_name	phenotype	pubmed_id


head outputs the first ten lines of files and with -1 it only outputs the first line (the header in this case).










## Problem 4

* Create a command that outputs the total number of lines in Locus_Germplasm_Phenotype_20130122.txt

In [10]:
wc -l Locus_Germplasm_Phenotype_20130122.txt

7216 Locus_Germplasm_Phenotype_20130122.txt


wc prints newline, word, and byte counts for each file. wc -l only prints the newline counts (7216 lines in this case).








## Problem 5

* Create a command that writes ONLY the data lines (i.e. excludes the header!) to a new file called "Data_Only.csv"
* prove that your output file has the expected number of lines

In [8]:
tail -7215 Locus_Germplasm_Phenotype_20130122.txt > Data_Only.csv
wc -l Data_Only.csv

7215 Data_Only.csv


As the file has 7216 lines (see Problem 4) and the header is in the first line, you can select the last 7215 lines with tail.


Using wc -l we can see that the new file has 7215 lines as expected.





## Problem 6

* Create a command that shows all of the lines that have a phenotype including the word "root"

In [9]:
grep "\sroot\s" -n Locus_Germplasm_Phenotype_20130122.txt

23:AT1G01550	bps1-2	Growth of bps1-2 mutants on CPTA-supplemented medium resulted in partial rescue of both leaf and root defects. bps1-2 mutants grown on control growth medium have small radialized leaves with very little vascular tissue, and very short misshapen knotted-looking roots. By contrast, bps1-2 mutants grown on CPTA-supplemented medium produced larger flattened leaves that contained primary and secondary veins, and smooth elongated roots.	17217459
24:AT1G01550	bps1-2	Mutant roots were abnormal; primary and lateral roots were short, root hairs formed close to the root apex, and root defects were most severe when plants were grown at low temperature.	15458645
26:AT1G01550	bps1-2	Under high light conditions (approximately 200 &mu;E.m-2.sec-1), CPTA-treated seedlings were completely photobleached and mutants showed rescue of both leaf and root development. Under moderate (approximately 100 &mu;E.m-2.sec-1) and low (30 &mu;E.m-2.sec-1) light intensities, seedlings are incomplete

255:AT1G08370	dcp1-2	The homozygous progeny is seedling lethal, showed arrested postembryonic development including cotyledon expansion, development of vascular networks, root elongation, and shoot development.	17485080
257:AT1G08430	Atalm1-KO	The mutant exhibits very low levels of root  growth and malate release in the presence of  Al (0.2 nmol plant-1 24 h-1).
289:AT1G09100	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf development and root elongation. The response is specific to glucose but not to osmotic changes.	17081979
294:AT1G09110	rpt5b	Insensitive to high-glucose repression of cotyledon expansion, chlorophyll accumulation, true-leaf development and root elongation. The response is specific to glucose but not to osmotic changes.	17081979
340:AT1G10130	eca3-1	Shoot and root growth is strongly retarded when plants are grown in the absence of Mn.  Also shows delayed growth when grown in the absence of Ca.	18024560
358:AT1G

675:AT1G22920	csn5a-1	Shorter hypocotyls in the dark, shorter  roots, smaller flowers, a decreased number of root hairs in response to jasmonic acid treatment, and altered light and auxin  responses.	17307927
687:AT1G22920	csn5a-2	Shorter hypocotyls in the dark, shorter  roots, smaller flowers, a decreased number of root hairs in response to jasmonic acid treatment, and altered light and auxin  responses.	17307927
701:AT1G23010	lpr1-1	Long primary root when grown on low Pi.	17496893
702:AT1G23010	lpr1-2	Long primary root when grown on low Pi.	17496893
707:AT1G23310	ggt1-1 (LUC)	In the absence of ABA treatment, root elongation of ggt1-1 was reduced nearly 50% compared to wild-type.	17318317
715:AT1G23320	tar1-1	This mutant does not have any obvious morphological defects and it responds normally to ACC and IAA in hypocotyl and root elongation assays.	18394997
792:AT1G26870	SALK_025663	Homozygotes have a reduced number of columella root cap (COL) and lateral root cap (LRC) cell layers. Me

1047:AT1G35580	cinv1	atinvg mutants have reduced root growth, reduced invertase activity, and increased expression of antioxidant genes under basal conditions.	21441406
1079:AT1G47056	vfb1-1/vfb2-1/vfb3-1/vfb4rnai	Defects in lateral root formation, delayed root elongation and delayed plant growth. Reduced cell size.	17435085
1080:AT1G47720	SALK_086929	Retarted root growth, variegated and distorted leaves, distorted flowers, partial sterile, and unviable seeds. Increased number but decreased size of mitochondria.	17189341
1081:AT1G47720	osb1-1	Retarted root growth, variegated and distorted leaves, distorted flowers, partial sterile, and unviable seeds. Increased number but decreased size of mitochondria.	17189341
1084:AT1G48380	rhl1-2	Extreme dwarf in the dark and in the light.  Reduced root hair number.  Trichome size and branching is reduced.  Ploidy defect.	16339310
1122:AT1G48920	parl1-1	parl1-1 mutants, initially identified based on their aberrant parallel venation pattern in juven

1322:AT1G59870	ko-1	shoot FW, root length and chlorophyll contents of ko-1 plants are significantly lower than those of wild-type plants	17355438
1323:AT1G59870	ko-2	shoot FW, root length and chlorophyll contents of ko-2 plants are significantly lower than those of wild-type plants	17355438
1328:AT1G60220	ots1-1 ots2-1	The ots1-1 ots2-1 double mutant plants flower earlier than wild type plants under long days and short days. They do not differ notably from wild type plants with respect to their overall growth rate, inflorescence structure, or seedling primary root elongation rate. But, they are hypersensitive to salt based on root elongation assays. They also have a higher level of SUMOylated proteins than wild type seedlings exposed to salt stress.	18849491
1334:AT1G60490	vps34	Similar to wild type siblings. Normal root hair  length. Segregates 1:1 for wt:het offspring as mutation affects the male gametophyte and is not transmitted.	18408046
1355:AT1G62990	knat7	No visible root phenot

1717:AT1G74030	eno1-1	Trichomes are less turgescent and are distorted with respect to the wild type.  Plants also have fewer root hairs with respect to wild type.	19223001
1718:AT1G74030	eno1-2	Trichomes are less turgescent and are distorted with respect to the wild type.  Plants also have fewer root hairs with respect to wild type.	19223001
1727:AT1G74660	35S::MIF1	Constitutive overexpression of MIF1 caused dramatic  developmental defects, including dwarfism, reduced apical dominance, extreme longevity, dark-green leaves,  altered flower morphology, poor fertility, reduced hypocotyl length, spoon-like cotyledons, reduced root  growth, and ectopic root hairs on hypocotyls and cotyledons. In addition, 35S::MIF1 seedlings underwent  constitutive photomorphogenesis in the dark, with root growth similar to that in the light. Furthermore,  35S::MIF1 seedlings were demonstrated to be non-responsive to gibberellin (GA) for cell elongation,  hypersensitive to the GA synthesis inhibitor paclobu

2007:AT2G01830	cre1-4	In the presence of kinetin, lateral root formation is not greatly inhibited, even at high concentrations.	12410813
2012:AT2G01830	stip-d ahk2-2 ahk3-3 cre1-12	Small SAM and limited root growth.	20110319
2013:AT2G01830	wol	Absence of periclinal cell divisions in the primary root meristem.	11114883
2016:AT2G01830	wol	Defect in cell division in the root and lower hypocotyl region soon after the embryo torpedo stage.	11114883
2019:AT2G01830	wol	Reduced cell number and exclusive xylem differentiation within the root vasculature.	11114883
2029:AT2G01830	wol-2	Rapidly accumulate anthocyanins due to the loss of root functionality.	15053761
2030:AT2G01830	wol-2	Short root phenotype.	15053761
2032:AT2G01830	wol-2	Viability depends on the ability to develop adventitious root before plant dies.	15053761
2037:AT2G01830	wol-3	Addition of IAA results in the formation of laterals from the primary root as is the case for wildtype. Both the laterals and primary root show retarded g

2468:AT2G27050	ebf1-3 ebf2-3 eil1-3	Modest rescue of the ebf1-3 ebf2-3 phenotype. Seedlings continue grow and develop at a slow pace after germination. Cotyledons partially expanded and turned pale green. Stunted root densely covered with root hair. Plants never flowered or set seeds.	17307926
2481:AT2G27228	SALK_079402 homozygous	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
2482:AT2G27230	SALK_079402 homozygous	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
2483:AT2G27230	lhw-1	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
2484:AT2G27230	lhw-2	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
2485:AT2G27230	lhw-3	Reduced root vascular population.  Roots lose bilateral symmetry and are monarch instead of diarch.	17626058
2486:AT2G27230	lhw-4	

2977:AT2G43790	mpk6-3	Defects in root development. Short roots and sometimes even no roots.Ectopic cell divisions in the roots and disorganized cells and cell files.	19484493
2985:AT2G44900	arabidillo-1/-2	Mutant plants grown on vertical plates had fewer lateral roots were developed than wild type or either single mutant, despite having similar primary root lengths to wild type. Significant decrease in number of lateral root primordia was found in double mutant roots, with no differences in later stages of lateral root development.	16434475
2992:AT2G44950	hub1-1	The mutant primary roots grew much more slowly than those of the wild type. No differences between mutant and WT were seen regarding cellular organization or cell wall formation in the root apical meristem. However, a clear reduction in meristem size was detected. The root growth rate of the mutant was severely reduced and no acceleration of growth took place. The average mature cortex cell length was significantly (P < 0.001) 

3304:AT3G05630	pldz2	mutant is defective in the hydrolysis of phospholipids and has a reduced capacity to accumulate galactolipids under limiting Pi conditions. Morphological analysis of the pldz2 root system shows a premature change in root architecture in response to Pi starvation.
3307:AT3G05630	pldzeta1 pldzeta2	Under limited phosphate, primary root elongation is retarded and lateral roots are longer with respect to wild type.	16384909
3308:AT3G06030	anp2 anp3	Delayed growth,short roots, multiply branched root hairs.Radially expanded root and hypocotyl cells.Aberrant microtuble organization and resistance to oryzalin.	20215588
3351:AT3G07780	obe1-1 obe2-1	Premature termination and loss of the shoot apical meristem. Failure to establish or maintain the root apical meristem.  Aberrant embryonic development.	18403411
3355:AT3G08040	CS6584	defective in iron translocation; chlorotic plants; expression of iron deficiency responses under conditions of iron sufficiency; overaccumulation of

3565:AT3G17360	pok1-1/pok2-1	At seedling stage pok1/pok2 double mutants differ from wild-type by having smaller cotyledons as well as shorter, wider roots and hypocotyls. Adult plants exhibit a dwarfed stature, but all organs are present, although miniaturized. Plants are fertile but produce reduced numbers of seeds. Microscopic observation revealed highly disordered root meristem cell patterns, with most walls lacking parallel or perpendicular alignment to the longitudinal axis of the root and most cells exhibiting irregular shapes. Examination of cell wall patterns suggested that misoriented cell divisions were largely responsible for the irregular cell patterns. In pok1/pok2 double mutant embryos, misplaced cell walls were observed as early as the octant stage of embryogenesis. Throughout embryogenesis, the mispositioning of cell walls continues, eventually giving rise to slightly misshapen mature embryos with a stubby appearance, compared to wild-type. However, incomplete cell wall

3881:AT3G28860	mdr1-1	Mutant mdr1-1 seedlings show an 80% reduction in acropetal auxin transport, but have normal levels of basipetal transport in their primary roots. Although gravitropism occurs normally in these mutants, they have a pronounced increase in root waviness resulting in a threefold increase in absolute curvature  relative to wild type seedlings.
3885:AT3G28860	mdr1-2	Lateral roots initiate and emerge at normal rates in mdr1-2 mutants, but they elongate ~50% more slowly than wild type lateral roots, and have a 60% reduction in acropetal auxin transport. Primary root elongation is not impaired in the mutants.	17557807
3886:AT3G28860	mdr1-2	Mutant mdr1-2 seedlings show an 80% reduction in acropetal auxin transport, but have normal levels of basipetal transport in their primary roots. Although gravitropism occurs normally in these mutants, they have a pronounced increase in root waviness resulting in a threefold increase in absolute curvature  relative to wild type seedlings

4191:AT3G55830	epc1-WiscDsLOX293-296invM1	Mutant plants have dramatically reduced growth habit, defects in vascular formation and reduced cell-cell adhesion properties in hypocotyl and cotyledon tissues. Secondary growth is enhanced in epc1 hypocotyl tissues. When germinated on MS agar plates, cotyledon and leaf expansion, primary root elongation and hypocotyl elongation of the mutant seedlings were reduced compared with the wild type. Epidermal cells of hypocotyls and roots exhibited a left-handed spiraled organization which was not present in the wild type. Cotyledons of epc1 were curled at the edges, thicker and more fragile than the wild type. Vascular patterning was disrupted in the epc1 mutants. Callose deposition in the vasculature of epc1 cotyledons was reduced compared to the wild type and the majority of seedlings showed no vascular-specific deposition. However, the majority of epc1 seedlings showed ectopic callose deposition seen as extensive punctate distribution of fluores

4639:AT4G14850	loi1-1	loi1-1 mutants are less sensitive to inhibitors of the MEP and MVA pathways of isoprenoid biosynthesis than wild type plants. Lovastatin (an inhibitor of HMG CoA reductase and the MVA pathway), normally inhibits root, growth alters root tip morphology, halts lateral root development, causes cotyledon contraction, and arrests true leaf development. But, all of these phenotypes are less prominent in loi1-1 mutants treated with lovastatin. Sterol levels are reduced less in loi1-1 mutants than in wild-type plants following treatment with lovastatin. HMG1 and HMG2 transcript levels are not altered in loi1-1 mutants, however, there is increased HMG CoA reductase activity in loi1-1 mutants compared to wild type when the plants are treated with lovastatin. Clomazone (an inhibitor of DXS and the MEP pathway) normally causes albinoism and defects in thylakoid development in wild type plants, but loi1-1 mutants treated with the same concentration of clomazone can develop gre

5007:AT4G31920	arr1-3 arr10-5 arr12-1	severe primary root abnormalities; premature termination of primary root growth; substantially reduced sensitivity to cytokinin in the hypocotyl growth response assay; smaller rosette diameter; altered chlorophyll and anthocyanin production.	18723577
5008:AT4G32040	SALK_000339	No visible root phenotype.	16463096
5028:AT4G32700	teb-1	A short-root phenotype with the phenotype in the aerial part that is similar to those of the tsk mutant. The roots in mutant seedlings were approximately one-third as long as those from wild type plants. This reduced growth became more pronounced at later time points. The root apical meristem was disorganized, and individual cell types, such as QC or initial cells, were difficult to discern. The aerial part of teb-1 plants also showed defects in morphology, such as abnormal phyllotaxy, highly serrated and asymmetric leaves and a fasciated stem. The cellular organization of the SAM was severely disturbed - L1 and L2 laye

5452:AT5G03540	exo70A1-1	Early seedling growth was not affected in mutants. Later, mutant seedlings had shorter roots, and retarded lateral rot intiation. The polar growth of root hairs and stigmatic papillae is disturbed. Organs are generally smaller, plants show a loss of apical dominance and indeterminate growth where instead of floral meristems new lateral inflorescences are initiated in a reiterative manner. Mutant plants had dramatically reduced fertility.	16942608
5453:AT5G03540	exo70a1-2	Early seedling growth was not affected in mutants. Later, mutant seedlings had shorter roots, and retarded lateral rot intiation. The polar growth of root hairs and stigmatic papillae is disturbed. Organs are generally smaller, plants show a loss of apical dominance and indeterminate growth where instead of floral meristems new lateral inflorescences are initiated in a reiterative manner. Mutant plants had dramatically reduced fertility.	16942608
5456:AT5G03730	CS8057	Mutation had dramatic effe

5668:AT5G10330	hpa1	Short roots. Initial development is normal but by 7 days post germination organization of the root apex begins to show abnormalities.	16815950
5672:AT5G10510	plt3-1	Short roots and shortened root meristem.	17960244
5673:AT5G10510	plt3-1/plt2-2/plt1-4	Rootless.Embryonic root differentiates 3 days after germination. Adventitious roots arrest 6 days after germination.	17960244
5680:AT5G11260	CS71	Mutants grown on agar plates have widely  spread lateral roots. Compared with that of wild-type plants, the direction of lateral root growth is nearly horizontal rather than downward, indicating alteration of the gravitropic response. The angles between the lateral roots and the orientation of gravity are larger than those of the wild type. In addition, secondary lateral roots grow and change their direction of elongation to the horizontal.	9367981
5686:AT5G11260	SALK_096651C	Decreased sensitivity to ABA induced lateral root inhibition. Increased sensitivity to osmotic and sal

5975:AT5G20490	WiscDsLox417C12 (HOMOZYGOUS)	There are no obvious defects in the aerial tissues of xi-k mutant plants grown under normal conditions. The root length of the mutant is also normal, but, it has shorter roots hairs than wild-type plants.	18178669
5976:AT5G20490	xi-k / xi-1	The xi-k / xi-1 double mutant plants are shorter and have smaller rosettes than wild type plants. Their reduced leaf size results from decreased cell size and decreased cell number. These double mutants produce fewer seeds per silique than wild type plants. In the leaf epidermis, Golgi stacks, peroxisomes, and mitochondria move more slowly in these mutant cells than in wild-type cells. Root hair length is reduced to ~35% of the wild type root hair length in these double mutants, but their root hair density is very similar to the wild type density.	19060218
5977:AT5G20490	xi-k / xi-2	The overall growth of the xi-k / xi-2 double mutant is normal. In the leaf epidermis, Golgi stacks, peroxisomes, and mitochon

6180:AT5G25220	SALK_136464	No visible root phenotype.	16463096
6186:AT5G25350	ebf1-1 ebf2-1	The mutant shows dwarfed growth, supernumerary epinastically curled leaves, early senescence, and abnormal flowers with the gynoecium protruding from the unopened floral buds. However, the phenotypes seen in double mutant plants are more severe than those seen in ctr1. This becomes apparent at about two weeks after germination, except for the reduced root growth, which is already obvious at three days after germination. Moreover, unlike the ctr1 mutation that has a reduced transmission of the ctr1 allele relative to wild-type and infertile early flowers, the double mutants have a greatly reduced fertility but can readily be pollinated with wild-type pollen.	14675533
6188:AT5G25350	ebf1-1 ebf2-1	When grown in the light, the double mutants display slightly longer hypocotyls, curled unexpanded cotyledons, and short and thicker roots with ectopic root hairs, similar to the ctr1 mutant. Moreover, the

6470:AT5G43940	GABI_315D11	Unable to grow following germination on nutrient medium in the dark. Unable to grow in the light on growth medium. The mutant seeds germinated, but growth was arrested right after some root elongation and emergence of small cotyledons. The small cotyledons failed to green and the seedlings eventually died. The mutants could however be recovered on soil. Under long day conditions (16 hour light), the mutants had a reduced number of rosette leaves. The leaves were pale green and distorted. Bolts were shorter. After bolting, the plants were highly branched and semi-dwarf under all light conditions. Few seed setting.	18326829
6487:AT5G44740	SALK_129731	Homozygotes are sensitive to UV-B treatment. UV-B treatment causes an inhibition of root growth.	18494853
6488:AT5G44750	SALK_005721	Mutant plants were moderately sensitive to UV-B irradiation than the wild type when mutant seedlings were irradiated with various doses of UV-B irradiation and grown in the light. Bot

6775:AT5G54440	club-2	Amorphous, club-like seedlings.  Reduced growth of root and shoot apices. Defects in cell plate assembly during cytokinesis.	20609115
6786:AT5G54690	irx8-1	Immunolabeling of stem and root sections using using monoclonal antibodies specific for glucuronoxylans revealed strong fluorescence signals in the walls of interfascicular fibers and xylem cells in wild-type Arabidopsis, but only weak signals were detected in the corresponding tissues of mutant plants.	17322407
6824:AT5G54690	irx8-6	Immunolabeling of stem and root sections using using monoclonal antibodies specific for glucuronoxylans revealed strong fluorescence signals in the walls of interfascicular fibers and xylem cells in wild-type Arabidopsis, but only weak signals were detected in the corresponding tissues of mutant plants.	17322407
6835:AT5G55480	shv3-2;svl1-1	Increased anthocyanin accumulation. Collapsed trichomes.Brown discoloration around endodermis in the hypocotyl and root due to ectopic lignific

7120:AT5G67270	eb1c-1	Plants grown on agar plants show leftward skewing of root growth occasionally forming clockwise spirals with twisted epidermal cell files.	18281505
7136:CUD	cud-1	When grown on media containing 0.5% sucrose, the mutant had reduced root length but no apparent radial cell expansion.	7743935
7137:DER2	der2	Root hairs have mostly very short root hairs, indicating a defect in the transition to rapid tip growth, which is required to fully develop the hair structure.	15863435
7138:DER5	der5	Root hairs display very short stumps or collapsed root hairs.  Root hairs are often wavy, with an irregular hair diameter.  Sometimes develop bulbous structures at the root hair base.	15863435
7139:DER6	der6	Root hairs display very short stumps or collapsed root hairs.  Root hairs have an irregular hair diameter or are collapsed.	15863435
7140:DER7	der7	Root hairs are very regular but significantly shorter than in the wild type.  Develops two different types of root hair structures: r

grep with -n prefixes each line of output with the 1-based line number within its input file to see all of the lines that have a phenotype including the word "root". I do not use -i because I do not want to ignore case distinctions.





## Problem 7

* Create a command that writes the AGI Locus Code for every line that has a phenotype including the word "root" to a file called: **Root-associated-Loci.txt**

In [7]:
grep "\sroot\s" -n Locus_Germplasm_Phenotype_20130122.txt | grep -iEo "AT[1-5]G[0-9]{5}" > Root-associated-Loci.txt

Using grep -o only prints the matched parts of a matching line. It allows to introduce all* the Locus Codes in a new file.


*Although the last loci do not follow the AGI Locus Code, I have only searched for AGI Locus Codes as required by the Problem.








## Problem 8

* Create a command that writes the PubMed ID for every line that has a phenotype including the word "root" to a file called: **Root-associated-Publications.txt**

In [6]:
grep "\sroot\s" -n Locus_Germplasm_Phenotype_20130122.txt | grep -iEo "\s[0-9]+$" > Root-associated-Publications.txt

As PubMed IDs are all digits (0-9) and the length is unpredictable, I use $ because it points out the end of a line.

## Problem 9

* _**Control experiment**_:  You would hypothesize that genes associated with roots **should be found on all chromosomes.**  Find a way (one or more commands) to test this hypothesis.  In this dataset, is the hypothesis true? 

In [8]:
echo "associated genes found on chromosome 1"
grep -iEo "AT[1]G[0-9]{5}" Root-associated-Loci.txt
echo "associated genes found on chromosome 2"
grep -iEo "AT[2]G[0-9]{5}" Root-associated-Loci.txt
echo "associated genes found on chromosome 3"
grep -iEo "AT[3]G[0-9]{5}" Root-associated-Loci.txt
echo "associated genes found on chromosome 4"
grep -iEo "AT[4]G[0-9]{5}" Root-associated-Loci.txt
echo "associated genes found on chromosome 5"
grep -iEo "AT[5]G[0-9]{5}" Root-associated-Loci.txt

associated genes found on chromosome 1
AT1G01550
AT1G01550
AT1G01550
AT1G02970
AT1G03000
AT1G03060
AT1G03060
AT1G03060
AT1G03060
AT1G04020
AT1G04030
AT1G04160
AT1G04160
AT1G04160
AT1G04220
AT1G04250
AT1G05180
AT1G05180
AT1G05850
AT1G05850
AT1G06290
AT1G06290
AT1G08030
AT1G08260
AT1G08370
AT1G08430
AT1G09100
AT1G09110
AT1G10130
AT1G10570
AT1G11130
AT1G15550
AT1G15690
AT1G16150
AT1G17170
AT1G17580
AT1G17580
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G19220
AT1G21700
AT1G22920
AT1G22920
AT1G23010
AT1G23010
AT1G23310
AT1G23320
AT1G26870
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27320
AT1G27760
AT1G28560
AT1G29990
AT1G31880
AT1G31880
AT1G31880
AT1G31880
AT1G32450
AT1G33280
AT1G33280
AT1G33290
AT1G33290
AT1G34370
AT1G34370
AT1G34370
AT1G35580
AT1G47056
AT1G47720
AT1G47720
AT1G48380
AT1G48920
AT1G49240
AT1G49240
AT1G49240
AT1G51190
AT1G52340
AT1G53700
AT1G53700
AT1G53700
AT1G54180
AT1G54960
AT1G54990
AT1G54990
A

AT3G43190
AT3G47450
AT3G47690
AT3G50080
AT3G50500
AT3G51160
AT3G51160
AT3G54140
AT3G54140
AT3G54140
AT3G55830
AT3G55830
AT3G55830
AT3G59420
AT3G59420
AT3G61650
AT3G61890
AT3G62980
AT3G63010
AT3G63010
AT3G63010
AT3G63010
AT3G63110
AT3G63110
AT3G63110
associated genes found on chromosome 4
AT4G17090
AT4G00710
AT4G01370
AT4G02780
AT4G05530
AT4G05530
AT4G05530
AT4G05530
AT4G05530
AT4G05530
AT4G07400
AT4G10310
AT4G10350
AT4G10350
AT4G14850
AT4G16760
AT4G18780
AT4G24400
AT4G24670
AT4G24670
AT4G24670
AT4G26080
AT4G26080
AT4G26690
AT4G26690
AT4G26690
AT4G26690
AT4G27630
AT4G29040
AT4G29040
AT4G29130
AT4G29130
AT4G29860
AT4G31920
AT4G32040
AT4G32700
AT4G32700
AT4G33470
AT4G33480
AT4G33495
AT4G33650
AT4G35460
AT4G36380
AT4G38130
AT4G38240
AT4G39350
associated genes found on chromosome 5
AT5G16150
AT5G01180
AT5G01410
AT5G01410
AT5G01410
AT5G01410
AT5G03150
AT5G03150
AT5G03280
AT5G03280
AT5G03280
AT5G03540
AT5G03540
AT5G03730
AT5G03730
AT5G03730
AT5G03730
AT5G03730
AT5G03860
AT5G03860
AT5G08020
AT

In this dataset, the hypothesis is true because there are AGI Locus Codes of the five chromosomes.

## Problem 10

* If your control experiment shows genes on every chromosome, then you can skip this question!  (you answered Problem 9 correctly!)

*  If your control experiment shows genes only on one or two chromosomes, then you have to explain why... what could the problem be?  (I told you specifically to be careful about this problem!)